dataplanes/vpp/pkg/nsmvpp/nsm_vpp_noimpl.go

```diff
@@ -7,19 +7,19 @@ import (
 	"github.com/ligato/networkservicemesh/pkg/nsm/apis/common"
 )
 
-type UnimplementedMechanism struct {
+type unimplementedMechanism struct {
 	Type common.LocalMechanismType
 }
 
 // CreateLocalConnect return error for unimplemented mechanism
-func (m UnimplementedMechanism) CreateLocalConnect(apiCh govppapi.Channel, srcParameters, dstParameters map[string]string) (string, error) {
-	return "", fmt.Errorf("%s mechanism not implemented", common.LocalMechanismType_name[int32(m.Type)])
+func (m unimplementedMechanism) createInterface(apiCh govppapi.Channel, parameters map[string]string) (uint32, error) {
```

Comment:
Would it make sense to do this with the LocalMechanism or RemoteMechanism object rather than just the parameters map?

Reply:
You are right, semantically more correct to go with Local/Remote Mechanisms. Technically at this point, Local/Remote mechanisms are just Type+Parameters, passing Type would be redundant, so until Local/RemoteMechanisms would introduce more information it's simpler to stay with parameters (as it's uniform type)